### PR TITLE
feat: REV-4051 | allow edx_django_service to be provided with specifi…

### PR DESCRIPTION
**Pull request by christopappas**
…c app names to run datadog on

---

Make sure that the following steps are done before merging:

  - [ ] Have a Site Reliability Engineer review the PR if you don't own all of the services impacted.
  - [ ] If you are adding any new default values that need to be overridden when this change goes live, update internal repos and add an entry to the top of the CHANGELOG.
  - [ ] Performed the appropriate testing.

### CODE REVIEW

File: playbooks/roles/edx_django_service/tasks/main.yml [hunk 1]
```diff
@@ -213,7 +213,9 @@
     - install:app-requirements
 
 - name: "Install Datadog APM requirements"
-  when: COMMON_ENABLE_DATADOG and COMMON_ENABLE_DATADOG_APP
+  when: >
+    COMMON_ENABLE_DATADOG and
+    (COMMON_ENABLE_DATADOG_APP or edx_django_service_name in edx_specifically_enabled_datadog_apps)
   pip:
```
Comment (christopappas) [inline playbooks/roles/edx_django_service/tasks/main.yml:217]: i'm no ansible expert, so, disclosure that i stole syntax from: https://github.com/edx/configuration/blob/01d773fbd5efaa0cd4616093794d1b8a25b84e65/playbooks/roles/user/tasks/main.yml#L261

**Comment by timmc-edx** [comment]
There's another pattern I've seen in this repo where edx_django_service computes a default and then allows it to be [overridden by services](https://github.com/edx/configuration/blob/ee5d9a301e44671119921a6a720f03f983f09782/playbooks/roles/ecommerce/meta/main.yml#L18) -- but it really gets quite verbose to chain it all through, so I like this list approach.

File: playbooks/roles/common_vars/defaults/main.yml [hunk 2]
```diff
@@ -102,6 +102,7 @@ COMMON_MONGO_READ_ONLY_PASS: !!null
 COMMON_ENABLE_DATADOG: False
 # Enable APM monitoring with Datadog (metrics, traces, and logs)
 COMMON_ENABLE_DATADOG_APP: False
+COMMON_ENABLE_SPECIFIC_DATADOG_APPS: []
```
Comment (timmc-edx) [inline playbooks/roles/common_vars/defaults/main.yml]: [suggestion] "App" in `COMMON_ENABLE_DATADOG_APP`  to application monitoring as opposed to infrastructure monitoring, so I'd rename to something like this:

```suggestion
COMMON_ENABLE_DATADOG_APP_SERVICES: []
```

...or something involving "services", maybe.

File: playbooks/roles/edx_django_service/defaults/main.yml [hunk 3]
```diff
@@ -104,6 +104,7 @@ edx_django_service_basic_auth_exempted_paths: '{{ edx_django_service_basic_auth_
 edx_django_service_newrelic_appname: '{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}-{{ edx_django_service_name }}'
 edx_django_service_enable_newrelic_distributed_tracing: false
 edx_django_datadog_service: 'edx-{{ edx_django_service_name }}'
+edx_specifically_enabled_datadog_apps: '{{ COMMON_ENABLE_SPECIFIC_DATADOG_APPS }}'
```
Comment (timmc-edx) [inline playbooks/roles/edx_django_service/defaults/main.yml]: If you lift the logic up to this as a default variable, then you can just use `{{ edx_django_service_datadog_enable }}` in the tasks (avoiding repetition of the logic).

```suggestion
edx_django_service_datadog_enable: '{{ COMMON_ENABLE_DATADOG and (COMMON_ENABLE_DATADOG_APP or edx_django_service_name in COMMON_ENABLE_DATADOG_APP_SERVICES }}'
```

Comment (timmc-edx) [reply playbooks/roles/edx_django_service/defaults/main.yml]: Here's another place you'd need to use that value: https://github.com/edx/configuration/blob/01d773fbd5efaa0cd4616093794d1b8a25b84e65/playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2#L23

File: playbooks/roles/edx_django_service/tasks/main.yml [hunk 4]
```diff
@@ -213,7 +213,9 @@
     - install:app-requirements
 
 - name: "Install Datadog APM requirements"
-  when: COMMON_ENABLE_DATADOG and COMMON_ENABLE_DATADOG_APP
+  when: >
+    COMMON_ENABLE_DATADOG and
+    (COMMON_ENABLE_DATADOG_APP or edx_django_service_datadog_enable)
```
Comment (timmc-edx) [inline playbooks/roles/edx_django_service/tasks/main.yml]: Now that we have the variable with a good default, we can just change this to consult that one variable:

```suggestion
  when: edx_django_service_datadog_enable
```

File: playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2 [hunk 5]
```diff
@@ -20,7 +20,7 @@ fi
 export NEW_RELIC_LICENSE_KEY="{{ NEWRELIC_LICENSE_KEY }}"
 {% endif -%}
 
-{% if COMMON_ENABLE_DATADOG and COMMON_ENABLE_DATADOG_APP %}
+{% if COMMON_ENABLE_DATADOG and (COMMON_ENABLE_DATADOG_APP or edx_django_service_datadog_enable) %}
```
Comment (timmc-edx) [inline playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2]: Same here:
```suggestion
{% if edx_django_service_datadog_enable %}
```

File: playbooks/roles/edx_django_service/tasks/main.yml (hunk 4 above)
Comment (timmc-edx) [reply playbooks/roles/edx_django_service/tasks/main.yml]: If we instead want to keep consulting the COMMON_ logic here, then the default should just be trimmed down to `edx_django_service_name in COMMON_ENABLE_DATADOG_APP_SERVICES`.

Comment (christopappas) [reply playbooks/roles/edx_django_service/tasks/main.yml]: duhhh sorry. yes. i was clearly not caffeinated enough this morning.

**Review by timmc-edx — APPROVED** [review]
Alright, let's see if it works! :crossed_fingers:

**Pull request merged**